src/test/e2e/api/admin/api-token.auth.e2e.test.ts

```diff
@@ -175,33 +176,32 @@ test('Token-admin should be allowed to create token', async () => {
 test('A role with only CREATE_PROJECT_API_TOKEN can create project tokens', async () => {
     expect.assertions(0);
 
-    const preHook = (app, config, { userService, accessService }) => {
+    const preHook = (
+        app,
+        config,
+        {
+            userService,
+            accessService,
+        }: { userService: UserService; accessService: AccessService },
+    ) => {
         app.use('/api/admin/', async (req, res, next) => {
-            const role = await accessService.getRootRole(RoleName.VIEWER);
+            const role = (await accessService.getRootRole(RoleName.VIEWER))!;
             const user = await userService.createUser({
                 email: 'powerpuffgirls_viewer@example.com',
                 rootRole: role.id,
             });
-            req.user = user;
             const createClientApiTokenRole = await accessService.createRole({
                 name: 'project_client_token_creator',
                 description: 'Can create client tokens',
-                permissions: [],
+                permissions: [{ name: CREATE_PROJECT_API_TOKEN }],
                 type: 'root-custom',
             });
-            await accessService.addPermissionToRole(
-                role.id,
-                CREATE_PROJECT_API_TOKEN,
-            );
             await accessService.addUserToRole(
                 user.id,
                 createClientApiTokenRole.id,
                 'default',
             );
-            req.user = await userService.createUser({
-                email: 'someguyinplaces@example.com',
-                rootRole: role.id,
-            });
+            req.user = user;
```

Comment:
On this side:
1. Get viewer role
2. Create a user with Viewer role as root role
3. Create a new role with CREATE_PROJECT_API_TOKEN permission
4. Add the user to the role created in `default` project
5. Associate `req.user` to the user created

Later we validate if this user can create an API token in the project where they have the role with that specific permission